bazel/markdown_to_html/src/tranformations/heading.ts

```diff
@@ -0,0 +1,41 @@
+/*!
+ * @license
+ * Copyright Google LLC All Rights Reserved.
+ *
+ * Use of this source code is governed by an MIT-style license that can be
+ * found in the LICENSE file at https://angular.dev/license
+ */
+
+import {RendererApi} from 'marked';
+
+import {getPageTitle} from '../utils';
+import {getHeaderId} from '../state';
+
+export const headingRender: RendererApi['heading'] = (text, level, raw) => {
+  if (level === 1) {
+    return `
+    <!-- Document Header -->
```

Comment:
Is the comment necessary? (admittedly small, but still extra bytes over the wire) 

If it is load bearing in some way, we should have a comment that calls that out 

(similar for other html comments)